model/storage-private/src/main/java/org/keycloak/storage/UserStorageManager.java

```diff
@@ -360,6 +360,8 @@ public boolean removeUser(RealmModel realm, UserModel user) {
             getFederatedStorage().preRemove(realm, user);
         }
 
+        publishUserPreRemovedEvent(realm, user);
```

Comment:
I think a better pattern is what we are using for group events. See https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/GroupModel.java#L67.

Not a blocker though as we can change that in a follow-up.